File: benchmarks/benchmarks/02_numerical_computation/comparisons/polynomial_eval.c
#include <stdio.h>

static long long polynomial_cost(long long points, long long degree) {
    long long checksum = 0;
    for (long long point = 0; point < points; ++point) {
        long long x = (point % 97) + 3;
        long long value = 1;
        for (long long coeff = degree; coeff > 0; --coeff) {
            value = ((value * x) + ((coeff * 11) + (point % 29))) % 1000003;
        }
        checksum += value;
    }
    return checksum;
}

int main(void) {
    printf("%lld\n", polynomial_cost(800000, 16));
    return 0;
}

File: benchmarks/benchmarks/02_numerical_computation/comparisons/polynomial_eval.rs
fn polynomial_cost(points: i64, degree: i64) -> i64 {
    let mut checksum = 0_i64;
    for point in 0..points {
        let x = (point % 97) + 3;
        let mut value = 1_i64;
        for coeff in (1..=degree).rev() {
            value = ((value * x) + ((coeff * 11) + (point % 29))) % 1_000_003;
        }
        checksum += value;
    }
    checksum
}

fn main() {
    println!("{}", polynomial_cost(800000, 16));
}
